icu4c/source/i18n/gregoimp.cpp

```diff
@@ -34,9 +34,14 @@ int64_t ClockMath::floorDivide(int64_t numerator, int64_t denominator) {
 
 int32_t ClockMath::floorDivide(double numerator, int32_t denominator,
                           int32_t& remainder) {
-    double quotient;
-    quotient = uprv_floor(numerator / denominator);
-    remainder = (int32_t) (numerator - (quotient * denominator));
+    // For an integer n and representable ⌊x/n⌋, ⌊RN(x/n)⌋=⌊x/n⌋, where RN is
+    // rounding to nearest.
+    double quotient = uprv_floor(numerator / denominator);
+    // For doubles x and n, where n is an integer and ⌊x+n⌋ < 2³¹, the
+    // expression `(int32_t) (x + n)` evaluated with rounding to nearest
+    // differs from ⌊x+n⌋ if 0 < ⌈x⌉−x ≪ x+n, as `x + n` is rounded up to
+    // n+⌈x⌉ = ⌊x+n⌋ + 1.  Rewriting it as ⌊x⌋+n makes the addition exact.
+    remainder = (int32_t) (uprv_floor(numerator) - (quotient * denominator));
```

Comment:
Same suggestion: I suggest combining the division and remainder into one operation.

Reply:
This function is THE combined ONE operation.